performant-software/DM
# Add highlight display toggle in text resources, plus viewer grid fixes

### What does this PR do?
- Adds to text resource frames a toggle button similar to the one in image frames to toggle the display of highlights on or off. When highlight display is turned off, highlights can still be made and only new highlights will be shown until the toggle is set to on.
- Fixes a vertical layering issue with the rollover/popup window
- Fixes various aspects of how viewer URIs and containers are tracked so that new additions appear in the correct place in the grid and with the correct read-only status.
- Changes wording of "lock for edit" text to "enable editing"
- If a resource title is changed to a blank string, it is reset to the pre-edit title upon edit completion

### What issues does this address?
Addresses #27, #71, #82, #86, #87, #88 

### How to test
Staging server: http://67.207.158.85/
- Open a text resource that has highlighted text. Click the eye icon in the toolbar to hide/show the highlights. With highlights hidden, create a new highlight by selecting text and pressing the tag icon. Turn highlights on and off again. In editing mode, try making changes to the text to make sure it still responds to the highlight toggle correctly.
- Open the rollover window several times. Resize it and reopen it. Confirm there's no visual weirdness (text showing through, misplaced resize handles, etc.)
- Add various text and image resources to the grid via annotations in the rollover window. Confirm that these resources added this way are always inserted immediately after the source document whose annotation brought up the rollover window. Confirm resources can be reordered via drag, closed and reopened, and that resources are only marked "read only" if the same resource is already in the grid.
- In editing mode, click the title of a resource to edit it. Delete the title, leaving it blank or with only space characters, and hit return / click outside the text field. The title should change back to its previous value rather than staying blank.
